### ISSUE TITLE
New tasmota, is this still needed?

### ISSUE THREAD

**Issue by beren12** [TheEebb/tasmota-tuyamcu-fix#3]
There's been some tuya mcu work done this winter according to the changelog. Is this branch still needed? I flashed my new dimmers with it but there are some nice features in 9.x that I'd love to have. Do you think you have time to update this, or possibly even make a PR so they can merge it? Thanks for the great work btw!

**Comment by beren12** [comment]
Looks like it is

**Comment by TheEebb** [comment]
I can certainly update this. However, the changes maybe too broad for a pull request. It really depends on whether they'll accept this as is. I don't have the time to strip it down to whatever their standards may be. The original idea was to make a robust module that could adapt to current and future Tuya MCU devices. However, it does not appear that things are going in that direction.

**Comment by beren12** [comment]
They did make some changes to the Tuya code recently but it doesn't fix the issue. I know they would at least look at your patch if you made a PR (easiest way is to click the fork button on their repo, commit the changes and click the "Make Pull Request" button

**Comment by beren12** [comment]
Add a writeup of what the changes do and I don't see them not merging it in some form at least.

**Comment by beren12** [comment]
> I can certainly update this. However, the changes maybe too broad for a pull request. It really depends on whether they'll accept this as is. I don't have the time to strip it down to whatever their standards may be. The original idea was to make a robust module that could adapt to current and future Tuya MCU devices. However, it does not appear that things are going in that direction.

From what I understand though it's trivial to replace the non-esp with an esp on most of the new devices so it's still useful to some people.

**Comment by digiblur** [comment]
> > I can certainly update this. However, the changes maybe too broad for a pull request. It really depends on whether they'll accept this as is. I don't have the time to strip it down to whatever their standards may be. The original idea was to make a robust module that could adapt to current and future Tuya MCU devices. However, it does not appear that things are going in that direction.
> 
> From what I understand though it's trivial to replace the non-esp with an esp on most of the new devices so it's still useful to some people.

Very much so, plus very addicting to do a transplant.

**Comment by TheEebb** [comment]
> Add a writeup of what the changes do and I don't see them not merging it in some form at least.

That's typically not how it works. The person submitting does all the work. Submitting a PR means you send it. Listen to feedback. Make the changes they request. Resubmit. If they don't like something- rinse and repeat. With a module and wholesale changes this can be time consuming.

Anyway, I dug into the new 9.3 code and significant changes were made to how communication with the web interface happens. Additionally RGB support was added. I decided to put some time into stripping down the existing 8.5 code and rewriting it for 9.3. This should make the code changes in 9.3 minimal and less intrusive. It should also make it simpler to submit a PR.

This means there will be a delay updating my build. I'm only able to sporadically work on this rewrite.

**Comment by beren12** [comment]
Any chance you'll have time to work on this soon? You seem to really understand the issues like nobody else.

**Comment by oldgeezy** [comment]
My FEIT Costco dimmers are only working with this branch.  With ESPHome and the latest version of Tasmota, eventually the MCU locks up and it stops responding to commands from the ESP module.  But then version 8.5.0.1 doesn't support MQTT auto-discovery (SetOption19 0) so they don't show up properly under the Tasmota integration.  I have no idea what's involved in revising this to a newer base version.  But you all would be doing me a huge favor if you could.

**Comment by btsimonh** [comment]
@oldgeezy - try:
https://github.com/btsimonh/Tasmota/wiki/Current-Binaries
It's 11 based, and resolves some Tuya issues related to the serial communications.
Be sure you can reflash to 8.5 before you upgrade (you may want to ensure you have a bench device with serial programming access - 8.5->11 was a pig for me).
And let me know?
(source is this branch: https://github.com/btsimonh/Tasmota/tree/TuyaUpdate)

**Comment by beren12** [comment]
While interesting, this code is 18months behind. I’ve been using v12 and it’s still broken I just deal with it atm. Any chance you are going to rebase it on the current code?

**Comment by btsimonh** [comment]
yeah, I've been a little lax on keeping my open source contributions up to date.  It would be interesting to know if it resolves the specific issues described here.

**Comment by brandondb1** [comment]
> My FEIT Costco dimmers are only working with this branch. With ESPHome and the latest version of Tasmota, eventually the MCU locks up and it stops responding to commands from the ESP module. But then version 8.5.0.1 doesn't support MQTT auto-discovery (SetOption19 0) so they don't show up properly under the Tasmota integration. I have no idea what's involved in revising this to a newer base version. But you all would be doing me a huge favor if you could.

I edited the 8.5 version's MQTT a bit to conform to the new MQTT standards in HASS - it seems to work fine for me (not Feit dimmers but seemingly very similar) and autodiscovers using MQTT and the older Home Assistant discovery (SetOption19 1).  You just need to change val_tpl to stat_val_tpl and compile it.

**Comment by btsimonh** [comment]
note recent updates to the tuyav2 driver
https://github.com/arendst/Tasmota/pull/20950
plus a couple of prior PRs - they relate to further dimmer issues.
(not my work - but building on what I did - but I've not tested with MY faulty tuya dimmers...).

**Comment by beren12** [comment]
Haven't tried v14 yet, but v13 still doesn't always turn the dimmers on in the morning or when I get home, but never seems to have issues turning off.
